src/games/oware.rs

```diff
@@ -0,0 +1,247 @@
+            }
+
+            if self.grand_slam(idx) {
+                return;
+            }
+
+            while match self.pits[idx] {
+                2 | 3 => self.is_in_opp_pits(idx),
+                _ => false,
+            } {
+                self.scores[player] += self.pits[idx];
+                self.pits[idx] = 0;
+                idx = (idx + PITS - 1) % PITS;
+            }
+        }
+
+        let draw = self.scores.iter().all(|&x| x == PITS as u8 * INIT_SEEDS / 2) || self.is_stalemate();
+
+        let won = self.scores[player] > PITS as u8 * INIT_SEEDS / 2;
+        self.outcome = if won {
+            Some(Outcome::WonBy(self.next_player))
+        } else if draw {
+            Some(Outcome::Draw)
+        } else {
+            None
+        };
+
+        self.next_player = self.next_player.other();
+    }
+
+    fn outcome(&self) -> Option<Outcome> {
+        self.outcome
+    }
+
+    fn can_lose_after_move() -> bool {
+        false
+    }
+}
+
+impl<const PITS: usize, const INIT_SEEDS: u8> Alternating for OwareBoard<PITS, INIT_SEEDS> {}
+
+impl<const PITS: usize, const INIT_SEEDS: u8> crate::board::BoardSymmetry<OwareBoard<PITS, INIT_SEEDS>>
+    for OwareBoard<PITS, INIT_SEEDS>
+{
+    type Symmetry = crate::symmetry::UnitSymmetry;
+    type CanonicalKey = ();
+
+    fn map(&self, _: Self::Symmetry) -> Self {
+        self.clone()
+    }
+
+    fn map_move(
+        &self,
+        _: Self::Symmetry,
+        mv: <OwareBoard<PITS, INIT_SEEDS> as Board>::Move,
+    ) -> <OwareBoard<PITS, INIT_SEEDS> as Board>::Move {
+        mv
+    }
+
+    fn canonical_key(&self) -> Self::CanonicalKey {}
+}
+
+impl<'a, const PITS: usize, const INIT_SEEDS: u8> BoardMoves<'a, OwareBoard<PITS, INIT_SEEDS>>
+    for OwareBoard<PITS, INIT_SEEDS>
+{
+    type AllMovesIterator = Internal<std::ops::Range<usize>>;
+    type AvailableMovesIterator = BruteforceMoveIterator<'a, OwareBoard<PITS, INIT_SEEDS>>;
+
+    fn all_possible_moves() -> Self::AllMovesIterator {
+        (0..PITS / 2).into_internal()
+    }
+
+    fn available_moves(&'a self) -> Self::AvailableMovesIterator {
+        BruteforceMoveIterator::new(self)
+    }
+}
+
+impl<const PITS: usize, const INIT_SEEDS: u8> Display for OwareBoard<PITS, INIT_SEEDS> {
+    fn fmt(&self, f: &mut Formatter<'_>) -> std::fmt::Result {
+        write!(f, "      ")?;
+        for i in 0..(PITS / 2) {
+            write!(f, " {i}   ")?;
+        }
+        writeln!(f, "     ")?;
+        write!(f, "+")?;
+        for _ in 0..(PITS / 2) + 2 {
+            write!(f, "----+")?;
+        }
+        writeln!(f)?;
+        write!(f, "│    │")?;
+        for i in 0..PITS / 2 {
+            write!(f, " {:2} │", self.pits[i])?;
+        }
+        writeln!(f, "    │")?;
+        write!(f, "│ {:2} +", self.scores[0])?;
+        for _ in 0..PITS / 2 {
+            write!(f, "----+")?;
+        }
+        writeln!(f, " {:2} │", self.scores[1])?;
+        write!(f, "│    │")?;
```

Comment:
It's a bit difficult to tell which score belongs to which player, would it be possible to offset them like this?

```
       0    1    2    3    4    5        
+----+----+----+----+----+----+----+----+
│    │  1 │  0 │  0 │  0 │  0 │  1 │ 25 │
│    +----+----+----+----+----+----+    │
│ 20 │  1 │  0 │  0 │  0 │  0 │  0 │    │
+----+----+----+----+----+----+----+----+
       5    4    3    2    1    0        
```
or even more clearly like this, so we also show which side is which:
```
       0    1    2    3    4    5        
+----+----+----+----+----+----+----+----+
│    │  1 │  0 │  0 │  0 │  0 │  1 │    │    B: 25
│    +----+----+----+----+----+----+    │
│    │  1 │  0 │  0 │  0 │  0 │  0 │    │    A: 20
+----+----+----+----+----+----+----+----+
       5    4    3    2    1    0        
```

If the current format is already an existing standard somewhere it's fine by me.